src/wallet/rpcwallet.cpp

```diff
@@ -5946,7 +5946,9 @@ UniValue z_shieldcoinbase(const UniValue& params, bool fHelp)
 #define MERGE_TO_ADDRESS_DEFAULT_TRANSPARENT_LIMIT 50
 #define MERGE_TO_ADDRESS_DEFAULT_SHIELDED_LIMIT 10
 
-#define JOINSPLIT_SIZE JSDescription::getNewInstance(GROTH_TX_VERSION).GetSerializeSize(SER_NETWORK, PROTOCOL_VERSION, GROTH_TX_VERSION)
+int GetJoinSplitSize(int shieldedTxVersion) {
+    return JSDescription::getNewInstance(shieldedTxVersion == GROTH_TX_VERSION).GetSerializeSize(SER_NETWORK, PROTOCOL_VERSION, shieldedTxVersion);
+}
```

Comment:
Just a small thing here; I'd suggest to put this function in an anonymous namespace (or alternatively make it `static`), so that its visibility remains limited to this file, given that it's not used elsewhere.

```suggestion
namespace {

int GetJoinSplitSize(int shieldedTxVersion) {
    return JSDescription::getNewInstance(shieldedTxVersion == GROTH_TX_VERSION).GetSerializeSize(SER_NETWORK, PROTOCOL_VERSION, shieldedTxVersion);
}

}
```

Reply:
Done